domains/transfer/Transfer.API/API.IntegrationTests/Claiming/Api/Controllers/ClaimControllerTest.cs

```diff
@@ -0,0 +1,132 @@
+using System.Threading.Tasks;
+using API.Claiming.Api.Dto.Response;
+using API.Claiming.Api.Models;
+using API.IntegrationTests.Factories;
+using API.IntegrationTests.Shared.Extensions;
+using FluentAssertions;
+using Xunit;
+
+namespace API.IntegrationTests.Claiming.Api.Controllers;
+
+public class ClaimControllerTest : IClassFixture<TransferAgreementsApiWebApplicationFactory>
+{
+    private readonly TransferAgreementsApiWebApplicationFactory factory;
+
+    public ClaimControllerTest(TransferAgreementsApiWebApplicationFactory factory)
+    {
+        this.factory = factory;
+    }
+
+    [Fact]
+    public async Task StopProcess_WhenClaimProcessExists_ReturnsNoContent()
+    {
+        var subject = Guid.NewGuid();
+        await factory.SeedClaims(new List<ClaimSubject>()
+        {
+            new(subject)
+        });
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.DeleteAsync("api/claim-automation/stop");
+        result.StatusCode.Should().Be(HttpStatusCode.NoContent);
+    }
+
+    [Fact]
+    public async Task StopProcess_WhenNoClaimProcessExists_ReturnsNotFound()
+    {
+        var subject = Guid.NewGuid();
+
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.DeleteAsync("api/claim-automation/stop");
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+
+    [Fact]
+    public async Task StartProcess_WhenNoClaimProcessHasStarted_ReturnsCreatedAt()
+    {
+        var subject = Guid.NewGuid();
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.PostAsync("api/claim-automation/start", null);
+        result.StatusCode.Should().Be(HttpStatusCode.Created);
+    }
+
+    [Fact]
+    public async Task StartProcess_WhenClaimProcessAlreadyStarted_ReturnsOk()
+    {
+        var subject = Guid.NewGuid();
+
+        await factory.SeedClaims(new List<ClaimSubject>()
+        {
+            new(subject)
+        });
+
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.PostAsync("api/claim-automation/start", null);
+        result.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenHistoryEntryHasDifferentSubjectId_ReturnsNotFound()
+    {
+        const string actor = "Peter Producent";
+        var client1 = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString(), actor: actor);
+        await client1.PostAsync("api/claim-automation/start", null);
+
+        var client = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString());
+        var result = await client.GetAsync("api/claim-automation/history");
+
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenHistoryExists_ReturnsOK()
```

Comment:
I would like that there is a history test for starting claim automation, stopping and then starting again. In that case, there should be 3 entries in the history response.

Reply:
Added